plasmapy/formulary/collisions.py

```diff
@@ -2033,3 +2038,91 @@ def coupling_parameter(
         )
 
     return coulomb_energy / kinetic_energy
+
+
+class CollisionFrequencies:
+    @validate_quantities(
+        v_a={"none_shall_pass": True},
+        T_a={
+            "none_shall_pass": True,
+            "can_be_negative": False,
+            "equivalencies": u.temperature_energy(),
+        },
+        n_b={"can_be_negative": False},
+        T_b={"can_be_negative": False, "equivalencies": u.temperature_energy()},
+    )
+    @particles.particle_input
+    def __init__(
+        self,
+        test_particle: particles.Particle,
+        field_particle: particles.Particle,
+        *,
+        v_a: u.cm / u.s = np.nan * u.cm / u.s,
```

Comment:
I think it would be easier to just set the default values for v_a and T_a to be None

Reply:
Hmm, the only issue with that would the replace_nan_velocities... Unless isnan() returns true for NaN

Reply:
Is there a problem with just doing something like: 

```python
if v_a is None and T_a is not None:
    v_a = thermal_speed(T_a, particle_a)
elif v_a is None and T_a is None:
    raise ValueError("One of v_a or T_a must be provided")
else:
   # Test to make sure v_a is an acceptable type input
```

Reply:
Yeah so currently the class is vectorized. In the case of a numpy array, if there is a mix of specified and unspecified v_a (unspecified v_a being represented by NaN) then v_a will be calculated from the corresponding T_a using the _replace_nan_velocities. By setting the default value to NaN, we mimic an entry for which the velocity is unspecified, and needs to be calculated using T_a

Reply:
Ah, I see, I don't think you need to support the use case where T_a is partially supplied and v_a is partially supplied. So, it could just be one or the other at a time.

Reply:
Ok that definitely simplifies things then! In that case we can use your suggested version for checking everything